lib/hammer_cli_foreman/associating_commands.rb

```diff
@@ -236,8 +236,42 @@ class AddProvisioningTemplateCommand < HammerCLIForeman::AddAssociatedCommand
         associated_resource :provisioning_templates
         desc _("Associate a provisioning template")
 
+        option "--provisioning-template-ids", "PROVISIONING_TEMPLATE_IDS", _("List of provisioning template ids"),
+               format: HammerCLI::Options::Normalizers::List.new
+        option "--provisioning-templates", "PROVISIONING_TEMPLATE_NAMES", _("List of provisioning template names"),
+               format: HammerCLI::Options::Normalizers::List.new
+        option "--provisioning-template-search", "PROVISIONING_TEMPLATE_SEARCH", _("Provisioning template name regex to search, all matching templates will be associate")
+
+        def request_params
+          params = super
+          template_ids = params['operatingsystem']['provisioning_template_ids']
+          if options['option_provisioning_template_ids']
+            template_ids |= options['option_provisioning_template_ids']
+          end
+          if options['option_provisioning_templates']
+            options['option_provisioning_templates']&.each do |provisioning_template_name|
+              template_id = HammerCLIForeman.collection_to_common_format(
+                associated_resource.action(:index).call(
+                  search: "name = \"#{provisioning_template_name}\"", per_page: 1000, page: 1)).first['id']
+              template_ids << template_id.to_s
+            end
+          end
+          if options['option_provisioning_template_search']
+            provisioning_templates = HammerCLIForeman.collection_to_common_format(
+              associated_resource.action(:index).call(per_page: 1000, page: 1)
+            )
+            matching_provisioning_templates = provisioning_templates.select do |ct|
+              ct['name'].match(options['option_provisioning_template_search'])
```

Comment:
Would it make sense to call api search in a similar way like you have for `--provisioning-templates`? The [~](https://github.com/wvanbergen/scoped_search/wiki/Query-language#comparison-operators) operator should do the job, but it is case-insensitive. If we want case-sensitive, I think it is ok to keep as is.